### PR TITLE
docs: terms and types doc for aa sdk v3 and other fixes

### PR DESCRIPTION
**Pull request by denniswon**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
The focus of this PR is to update various documentation files and type definitions related to smart account clients and user operations.

### Detailed summary
- Updated documentation and type definitions for smart account clients and user operations
- Added references and links to external resources for better understanding
- Clarified optional parameters and override values for user operations

> The following files were skipped due to too many changes: `site/snippets/aa-core/send-tx-param.md`, `site/packages/aa-core/smart-account-client/actions/signUserOperation.md`, `site/smart-accounts/light-account/index.md`, `site/packages/aa-core/smart-account-client/actions/dropAndReplaceUserOperation.md`, `site/snippets/aa-core/send-uo-param.md`, `site/packages/aa-core/smart-account-client/actions/sendTransactions.md`, `site/packages/aa-core/smart-account-client/actions/buildUserOperationFromTxs.md`, `site/packages/aa-core/smart-account-client/actions/buildUserOperation.md`, `site/getting-started/setup.md`, `site/packages/aa-core/smart-account-client/actions/sendUserOperation.md`, `site/packages/aa-core/smart-account-client/actions/checkGasSponsorshipEligibility.md`, `site/resources/faqs.md`, `site/packages/aa-core/smart-account-client/middleware/index.md`, `site/packages/aa-core/smart-account-client/types/userOperationOverrides.md`, `packages/core/src/types.ts`, `site/resources/terms.md`, `site/resources/types.md`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/core/src/types.ts [hunk 1]
```diff
@@ -134,60 +135,7 @@ export interface UserOperationReceipt {
...
-  root: Hex;
-  /* Either 1 (success) or 0 (failure). (post Byzantium hard fork at block 4,370,000) */
-  status: number;
-  /* The cumulative gas used in the block containing this UserOperation. */
-  effectiveGasPrice: BigNumberish;
-  /* The type of the recipt object */
-  type: string;
-}
-
-export interface UserOperationReceiptLog {
```
Comment (denniswon) [inline packages/core/src/types.ts:172]: this is same as viem's EventLog.

File: packages/core/src/types.ts [hunk 2]
```diff
@@ -134,60 +135,7 @@ export interface UserOperationReceipt {
   /* The logs generated by this UserOperation (not including logs of other UserOperations in the same bundle). */
   logs: string[];
   /* The TransactionReceipt object for the entire bundle, not only for this UserOperation. */
-  receipt: UserOperationReceiptObject;
-}
-
-export interface UserOperationReceiptObject {
```
Comment (denniswon) [inline packages/core/src/types.ts:140]: this is same as viem's TransactionReceipt

File: site/getting-started/setup.md [hunk 3]
```diff
@@ -154,20 +154,18 @@ When running the above script, you might see the following errors:
 These are due to increased network activity at that time and are fleeting issues. Running the script again will resolve them naturally.
 :::
 
-Since this "Getting started" example is a simple script, you must consider how Account Kit can work in various applications. Check out our [Demos](/overview/demos) to see how.
+Since this "Getting started" example is a simple script, you must consider how Account Kit can work in various applications. {{ /* Check out our [Demos](/overview/demos) to see how. */ }}
```
Comment (denniswon) [inline site/getting-started/setup.md:157]: commented out the lines to check out our demo for now.

File: site/getting-started/setup.md [hunk 4]
```diff
@@ -154,20 +154,18 @@ When running the above script, you might see the following errors:
...
+3. To learn more about different `signer` options for your smart accounts, check out the section on [Choosing a signer](/signers/choosing-a-signer).
 
-4. To learn more about different User Operations, you can send with different `target` and `data` fields in the `sendUserOperation` function above, look at our [How to send a User Operation](/using-smart-accounts/send-user-operations) guide for an example using NFT mints.
+4. To learn more about how to use your smart accounts and what Account Kit offers to enhance users' web3 experience, check out a number of guides we have in the [Using smart accounts](/using-smart-accounts/send-user-operations.md) section, covering from basic to advanced usages.
 
-5. To explore more ways to use Account Kit, check out the step-by-step guides, such as [How to sponsor gas for a UserOperation](/using-smart-accounts/sponsoring-gas/gas-manager) or [How to fetch a Smart Account's NFTs](/using-smart-accounts/enhanced-apis/nft).
-
-6. To see Account Kit in action, check out our [Demos](/overview/demos).
+{{ /* 5. To see the Account Kit in action, check out our [Demos](/overview/demos). */ }}
```
Comment (denniswon) [inline site/getting-started/setup.md:171]: ditto: commented out the lines to check out our demo for now.

File: packages/core/src/types.ts (hunk 2 above)
Comment (moldy530) [reply packages/core/src/types.ts:140]: The issue with deleting these types is if anyone was importing them before they now would end up with a breaking change

Comment (denniswon) [reply packages/core/src/types.ts:142]: got it. will put it back but put deprecated tag

**Review by moldy530 — APPROVED** [review]

**Pull request merged**